docs/sc-memory/api/cpp/extended/permissions_api.md

```diff
@@ -0,0 +1,809 @@
+# **User permissions API**
+
+!!! warning
+    This documentation is correct for only versions of sc-machine that >= 0.10.0.
+---
+
+This API allows to handle users and their permissions in knowledge base.
+
+## **Who are users of ostis-systems?**
+
+Users interact with ostis-systems. Users can be other systems, agents or people. All users of an ostis-system can perform actions on its knowledge base. There are six classes of user actions in knowledge base. Before a user action is  performed, it is checked whether the user has permissions to perform actions of the specified class.
+
+By default, user permissions are not handled by sc-machine. This is configured via the sc-machine config (`<config-name>.ini`). To enable user permissions handling in the sc-machine, go to the group `[sc-memory]` and set `user_mode` as `true`.
+
+```ini
+[sc-memory]
+...
+user_mode = true
+...
+```
+
+## **How are users managed?**
+
+You can only work with memory through methods provided in `ScMemoryContext`. Each object of this class can be considered as an object that handles information about a user (including his permissions) when that user invokes methods through that object.
+
+You can't create object of `ScMemoryContext` with providing user. This happens automatically. When some user initiates a sc-event, an object of `ScMemoryContext` with this user is created for the agent that reacted to this sc-event. After your agent uses this context to call sc-memory methods.
```

Comment:
unclear explanation. How can a user create an event if he doesn't have a context yet? Info about guest user should be above

Reply:
let's discuss

Reply:
Fixed